user/applications/tinker/src/fqc_test.cpp

```diff
@@ -486,27 +495,191 @@ bool FqcTest::wifiScanNetworks(JSONValue req) {
-    writer.beginObject();
-    writer.name("pass").value(true);
-    writer.name("networks").beginArray();
+    writer_.beginObject();
+    writer_.name("pass").value(true);
+    writer_.name("networks").beginArray();
     for(WiFiAccessPoint& network: networks){
-        writer.beginObject();
-        writer.name("ssid").value(String(network.ssid, network.ssidLength));
+        writer_.beginObject();
+        writer_.name("ssid").value(String(network.ssid, network.ssidLength));
 
         char bssidStr[32] = {};
         sprintf(bssidStr, "%02X:%02X:%02X:%02X:%02X:%02X", network.bssid[0], network.bssid[1], network.bssid[2], network.bssid[3], network.bssid[4], network.bssid[5]);
-        writer.name("bssid").value(String(bssidStr));
+        writer_.name("bssid").value(String(bssidStr));
 
-        writer.name("security").value((int)network.security);
-        writer.name("channel").value(network.channel);
-        writer.name("rssi").value(network.rssi);
-        writer.endObject();
+        writer_.name("security").value((int)network.security);
+        writer_.name("channel").value(network.channel);
+        writer_.name("rssi").value(network.rssi);
+        writer_.endObject();
     }
-    writer.endArray();
-    writer.endObject();
+    writer_.endArray();
+    writer_.endObject();
     return true;
 }
 
+#if PLATFORM_ID == PLATFORM_MSOM
+static int callbackGPSGSV(int type, const char* buf, int len, FqcTest* self) {
+    // EXAMPLE:
+    // $<TalkerID>GSV,<TotalNumSen>,<SenNum>,<TotalNumSat>
+    //  {,<SatID>,<SatElev>,<SatAz>,<SatCN0>},
+    // <SignalID>*<Checksum><CR><LF>
+
+    // Single sat:
+    // > AT+QGPSGNMEA="GSV"
+    // < +QGPSGNMEA: $GPGSV,1,1,01,30,,,32,1*67
+    // < OK
+    
+    // Multiples:
+    // > AT+QGPSGNMEA="GSV"
+    // < +QGPSGNMEA: $GPGSV,3,1,11,30,46,293,32,04,23,134,00,05,11,319,00,07,68,346,00,1*6B
+    // < +QGPSGNMEA: $GPGSV,3,2,11,08,37,096,00,09,61,141,00,14,33,221,00,16,01,042,00,1*68
+    // < +QGPSGNMEA: $GPGSV,3,3,11,20,23,285,00,22,13,218,00,27,21,053,00,1*51
+    // < +QGPSGNMEA: $GLGSV,3,1,09,78,59,010,00,86,,,00,77,08,046,00,80,01,239,00,1*44
+    // < +QGPSGNMEA: $GLGSV,3,2,09,79,48,263,00,69,44,322,00,88,02,103,00,87,08,055,00,1*7E
+    // < +QGPSGNMEA: $GLGSV,3,3,09,67,24,162,00,1*43
+    // < OK
+
+    // GL = Glonass
+    // GP = GPS
+    // PQ = BeiDou
+    // GA = Galileo
+    
+    //Log.trace("%d : %s", strlen(buf), buf);
+
+    // If this is the trailing OK response, exit
+    if(!strcmp(buf, "\r\nOK\r\n")) {
+        return 0;
+    }
+
+    const int MAX_GSV_STR_LEN = 128;
+    char gsvSentence[MAX_GSV_STR_LEN] = {};
+    strlcpy(gsvSentence, buf, MAX_GSV_STR_LEN);
+    
+    const char * delimiters = ", ";
+    char * token = strtok(gsvSentence, delimiters);
+    int i = 1;
+    while (token) {
+        //Log.trace("%d %s", i, token);
+        token = strtok(NULL, delimiters);
+        i++;
+
+        switch (i) {
+            case 5: // TotalNumSat 
```

Comment:
There are two constellations reported in the GSV sentences.  This method may take the last set only, which is GLONASS.

Reply:
Yes, this just lazily finds the last GSV sentence satellite count. I didnt do any more parsing since its not clear what data (if any) we care about returning to the host. 

If we can see any number of satellites, then I consider gnss working and return success. 
If we want this test to do more, then we need to specify that and I can improve it.

Reply:
You may want to target the constellation that would be provided by the GNSS simulator during manufacturing

Reply:
Eugene posted an example simulator that reported GPS, but we dont know for sure what the factory will end up using. This will work with any constellation. 

If we care about detecting/parsing each individual constellation I can do that, but we should make sure this will be used somewhere.